cluster-autoscaler/cloudprovider/packet/README.md

```diff
@@ -0,0 +1,53 @@
+# Cluster Autoscaler for Packet
+
+The cluster autoscaler for [Packet](https://packet.com) worker nodes performs
+autoscaling within any specified nodepool. It will run as a `Deployment` in
+your cluster. The nodepool is specified using tags on Packet.
+
+This README will go over some of the necessary steps required to get
+the cluster autoscaler up and running.
+
+## Permissions and credentials
+
+The autoscaler needs a `ServiceAccount` with permissions for Kubernetes and
+requires credentials for interacting with OpenStack.
+
+An example `ServiceAccount` is given in [examples/cluster-autoscaler-svcaccount.yaml](examples/cluster-autoscaler-svcaccount.yaml).
+
+The credentials for authenticating with Packet are stored in a secret and
+provided as an env var to the container. [examples/cluster-autoscaler-secret](examples/cluster-autoscaler-secret.yaml)
+In the above file you can modify the following fields:
+
+| Secret                          | Key                | Value                        |
+|---------------------------------|--------------------|------------------------------------------------------------------------------------------------------------------------------------|
+| cluster-autoscaler-packet       | authtoken          | Your Packet API token                                                                                                              |
+| cluster-autoscaler-cloud-config | Global/project-id  | Your Packet project id                                                                                                             |
+| cluster-autoscaler-cloud-config | Global/api-server  | The ip:port for you cluster's k8s api (e.g. K8S_MASTER_PUBLIC_IP:6443)                                                             |
+| cluster-autoscaler-cloud-config | Global/facility    | The Packet facility for the devices in your nodepool (default: ams1)                                                               |
+| cluster-autoscaler-cloud-config | Global/plan        | The Packet plan (aka size/flavor) for new nodes in the nodepool (default: t1.small)                                                |
+| cluster-autoscaler-cloud-config | Global/billing     | The billing interval for new nodes (default: hourly)                                                                               |
+| cluster-autoscaler-cloud-config | Global/os          | The OS image to use for new nodes (default: ubuntu_18_04). If you change this also update cloudinit.                               |
+| cluster-autoscaler-cloud-config | Global/cloudinit   | The base64 encoded init script that installs Docker and kubeadm on newly provisioned nodes. Default value tested with Ubuntu 18.04 |
```

Comment:
Hm, rather than specifying the script here, could we use cloud-init config here? Or is it both supported, as in https://support.packet.com/kb/articles/user-data ? If so, then description here could be different, since in our case we will use cloud-init format for provisioning new nodes.